### ISSUE TITLE
Missing required architecture arm64

### ISSUE THREAD

**Issue by samuelpiltch** [nitaliano/react-native-mapbox-gl#567]
While trying to install my React Native app with Mapbox API onto my iPhone 6S, I receive the following error: 

> ld: warning: ignoring file /Users/sam/GitHub/walk-to-mobile/ios/Mapbox.framework/Mapbox, missing required architecture arm64 in file /Users/sam/GitHub/walk-to-mobile/ios/Mapbox.framework/Mapbox (2 slices)
Undefined symbols for architecture arm64:
  "_MGLOfflinePackErrorNotification", referenced from:
      -[RCTMapboxGLManager init] in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_MGLOfflinePackErrorUserInfoKey", referenced from:
      -[RCTMapboxGLManager offlinePackDidReceiveError:] in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_MGLOfflinePackMaximumCountUserInfoKey", referenced from:
      -[RCTMapboxGLManager offlinePackDidReceiveMaximumAllowedMapboxTiles:] in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_MGLOfflinePackMaximumMapboxTilesReachedNotification", referenced from:
      -[RCTMapboxGLManager init] in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_MGLOfflinePackProgressChangedNotification", referenced from:
      -[RCTMapboxGLManager init] in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLAccountManager", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLAnnotationImage", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGL.o)
  "_OBJC_CLASS_$_MGLAnnotationView", referenced from:
      _OBJC_CLASS_$_RCTMapboxAnnotation in libRCTMapboxGL.a(RCTMapboxAnnotation.o)
  "_OBJC_CLASS_$_MGLMapCamera", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLMapView", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGL.o)
  "_OBJC_CLASS_$_MGLMultiPointFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLMultiPolygonFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLMultiPolylineFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLOfflineStorage", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLPointFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLPolygon", referenced from:
      _OBJC_CLASS_$_RCTMGLAnnotationPolygon in libRCTMapboxGL.a(RCTMapboxGL.o)
      l_OBJC_$_CATEGORY_MGLPolygon_$_RCTAdditions in libRCTMapboxGL.a(MGLPolygon+RCTAdditions.o)
  "_OBJC_CLASS_$_MGLPolygonFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLPolyline", referenced from:
      _OBJC_CLASS_$_RCTMGLAnnotationPolyline in libRCTMapboxGL.a(RCTMapboxGL.o)
      l_OBJC_$_CATEGORY_MGLPolyline_$_RCTAdditions in libRCTMapboxGL.a(MGLPolyline+RCTAdditions.o)
  "_OBJC_CLASS_$_MGLPolylineFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLShapeCollectionFeature", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLStyle", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_CLASS_$_MGLTilePyramidOfflineRegion", referenced from:
      objc-class-ref in libRCTMapboxGL.a(RCTMapboxGLManager.o)
  "_OBJC_METACLASS_$_MGLAnnotationView", referenced from:
      _OBJC_METACLASS_$_RCTMapboxAnnotation in libRCTMapboxGL.a(RCTMapboxAnnotation.o)
  "_OBJC_METACLASS_$_MGLPolygon", referenced from:
      _OBJC_METACLASS_$_RCTMGLAnnotationPolygon in libRCTMapboxGL.a(RCTMapboxGL.o)
  "_OBJC_METACLASS_$_MGLPolyline", referenced from:
      _OBJC_METACLASS_$_RCTMGLAnnotationPolyline in libRCTMapboxGL.a(RCTMapboxGL.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

The app builds successfully onto the iPhone simulator. I am currently running the latest version of Mapbox, with React Native 0.44.2. I have already checked if the architecture is missing with lipo, getting the following:

> $ lipo ~/Github/walk-to-mobile/ios/Mapbox.framework/Mapbox -info
> Architectures in the fat file: /Users/sam/Github/walk-to-mobile/ios/Mapbox.framework/Mapbox are: i386 x86_64 armv7 arm64 

This means that the architecture is there, but Xcode cannot find it. Has anyone else run into the same problem or know a fix?
Thanks, will be glad to answer any questions.

**Comment by henrikra** [comment]
Solved?

**Comment by samuelpiltch** [comment]
Xcode bug. Reinstalling the MapBox framework fixed this issue for me.

**Comment by iamtommcc** [comment]
I also experienced this issue on Xcode 10/Mac OS Mojave on Mapbox v 6.1.3. I tried...

* Downgrading Mapbox versions
* Completing removing & unlinking Mapbox, then relinking from scratch (tried this several times)
* Tweaking every Xcode build setting imaginable that is even remotely relevant
* Changing the version of the Mapbox iOS SDK that the react native wrapper uses
* Toggling between Xcode default build system and "legacy" build system

No luck!

A very odd issue, since at one point it was working just fine.
